packages/reports/src/Containers/open-positions.tsx

```diff
@@ -115,98 +101,63 @@ type TTotals = {
     payout?: number;
 };
 
-type TAddToastProps = {
-    key?: string;
-    content: string;
-    timeout?: number;
-    is_bottom?: boolean;
-    type?: string;
-};
-
-type TOpenPositions = Pick<
-    TPortfolioStore,
-    | 'active_positions'
-    | 'error'
-    | 'getPositionById'
-    | 'is_loading'
-    | 'is_multiplier'
-    | 'onClickCancel'
-    | 'onClickSell'
-    | 'onMount'
-> & {
+type TOpenPositions = {
     component_icon: string;
-    currency: string;
-    is_accumulator: boolean;
-    is_eu: boolean;
-    is_virtual: boolean;
-    NotificationMessages: () => JSX.Element;
-    server_time: moment.Moment;
-    addToast: (obj: TAddToastProps) => void;
-    current_focus: string;
-    onClickRemove: () => void;
-    getContractById: (contract_id?: number) => TContractStore;
-    removeToast: () => void;
-    setCurrentFocus: () => void;
-    should_show_cancellation_warning: boolean;
-    toggleCancellationWarning: () => void;
-    toggleUnsupportedContractModal: () => void;
 };
 
-type TMobileRowRendererProps = Pick<
-    TOpenPositions,
-    | 'addToast'
-    | 'current_focus'
-    | 'getContractById'
-    | 'onClickRemove'
-    | 'removeToast'
-    | 'setCurrentFocus'
-    | 'should_show_cancellation_warning'
-    | 'toggleCancellationWarning'
-    | 'toggleUnsupportedContractModal'
-> &
-    Omit<TMobileRowRenderer, 'columns_map'> & {
-        columns_map: { [key: TColIndex]: undefined | TDataListCell['column'] };
-    };
+const EmptyPlaceholderWrapper = ({ is_empty, component_icon, children }: TEmptyPlaceholderWrapper) => (
+    <React.Fragment>
+        {is_empty ? (
+            <PlaceholderComponent
+                is_empty={is_empty}
+                empty_message_component={EmptyTradeHistoryMessage}
+                component_icon={component_icon}
+                localized_message={localize('You have no open positions yet.')}
+            />
+        ) : (
+            children
+        )}
+    </React.Fragment>
+);
 
 const MobileRowRenderer = ({
-    row = {},
+    row,
     is_footer,
-    columns_map = {},
-    server_time = toMoment(),
+    columns_map,
+    server_time,
     onClickCancel,
     onClickSell,
     measure,
     ...props
-}: TMobileRowRendererProps) => {
+}: TMobileRowRenderer) => {
     React.useEffect(() => {
         if (!is_footer) {
             measure?.();
         }
-    }, [row.contract_info?.underlying, measure, is_footer]);
+    }, [row?.contract_info?.underlying, measure, is_footer]);
 
     if (is_footer) {
         return (
             <>
                 <div className='open-positions__data-list-footer--content'>
                     <div>
-                        <DataList.Cell row={row} column={columns_map.purchase} />
-                        <DataList.Cell row={row} column={columns_map.payout} />
+                        <DataList.Cell row={row} column={columns_map?.purchase} />
```

Comment:
@henry-deriv no need for question marks after columns_map on these lines, let's keep default values for the props above

Reply:
changed